packages/python/plotly/plotly/express/_core.py

```diff
@@ -1162,7 +1164,7 @@ def get_orderings(args, grouper, grouped):
                 key=lambda g: orders[col].index(g[i]) if g[i] in orders[col] else -1,
             )
 
-    return orders, group_names
+    return orders, group_names, group_values
```

Comment:
what is the difference between `orders` and `group_values`? In the examples I tried (including with facets and user-specified category orders) they were the same (the keys were not in the same order, but keys of dicts are not ordered I think?)

Reply:
`orders` includes all values given by the user even if they don't show up in the data

Reply:
whereas `group_values` has keys only per-group and has only values that show up in the data, so it's more specific. I'm pretty sure we need both, or at least I'm not comfortable trimming down `orders` at the moment :)

Reply:
Thanks. Would you mind adding a comment? The number of variables exceeds by working memory :-)

Reply:
pushed!

Reply:
thanks